### PR TITLE
Add SMS MFA management custom flow

### PR DESCRIPTION
**Pull request by royanger**
> [!IMPORTANT]
> 🔎 Previews:
>
> - https://clerk.com/docs/pr/1195/custom-flows/manage-sms-based-mfa

### Explanation: 
Custom flow code for managing SMS based multi-factor authentication

### This PR:
Documentation was missing

### CODE REVIEW

File: docs/custom-flows/manage-sms-based-mfa.mdx [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+    const [code, setCode] = React.useState('');
+    const [isVerifying, setIsVerifying] = React.useState(false);
+    const [successful, setSuccessful] = React.useState(false);
+    const [phoneObj, setPhoneObj] = React.useState<
+      PhoneNumberResource | undefined
+    >();
+
+    if (!isLoaded) return null;
+
+    if (isLoaded && !user?.id) {
```
Comment (royanger) [inline docs/custom-flows/manage-sms-based-mfa.mdx:261]:
```suggestion
    if (!user?.id) {
```
With the above check for `!isLoaded` this should be redundant.

File: docs/custom-flows/manage-sms-based-mfa.mdx [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+    // Check if any phone numbers are reserved for MFA
+    const mfaPhones = user.phoneNumbers
+      .filter((ph) => ph.verification.status === 'verified')
+      .filter((ph) => ph.reservedForSecondFactor)
+      .sort((ph: PhoneNumberResource) => (ph.defaultSecondFactor ? -1 : 1));
+
+    return (
+      <>
+        <h2>Phone numbers reserved for MFA</h2>
+        {mfaPhones.length === 0 ? (
```
Comment (royanger) [inline docs/custom-flows/manage-sms-based-mfa.mdx:62]: I don't like ternaries used this way. I would prefer this get reverted to a separate return for the 'no number' state, like:
```
if (mfaPhones.length === 0) {
  return <p>No phone numbers configured for MFA</p>;
}

File: docs/custom-flows/manage-sms-based-mfa.mdx [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+      // Set the phone number as reserved for MFA
+      await phone.setReservedForSecondFactor({ reserved: true });
+      // Refresh the user information to reflect changes
+      await user.reload();
+    };
+
+    return (
+      <>
+        <h2>Phone numbers that are not reserved for MFA</h2>
+        {availalableForMfaPhones.length === 0 ? (
```
Comment (royanger) [inline docs/custom-flows/manage-sms-based-mfa.mdx:126]: Same request for the separate return here.

File: docs/custom-flows/manage-sms-based-mfa.mdx [hunk 4]
```diff
@@ -0,0 +1,374 @@
...
+          <p>
+            None - <Link href="/account/add-phone">Add a new phone number</Link>
```
Comment (royanger) [inline docs/custom-flows/manage-sms-based-mfa.mdx:128]: A button to add a number should always be present. Once a user adds one number, this button should no longer show. Let's move this back to the bottom.

File: docs/custom-flows/manage-sms-based-mfa.mdx [hunk 5]
```diff
@@ -0,0 +1,374 @@
...
+  }
+
+  export default function ManageSMSMFA() {
+    const [showBackupCodes, setShowBackupCodes] = React.useState(false);
+
+    const { isLoaded, user } = useUser();
+
+    if (!isLoaded) return null;
+
+    if (isLoaded && !user) {
```
Comment (royanger) [inline docs/custom-flows/manage-sms-based-mfa.mdx:208]:
```suggestion
    if (!user) {
```
As mentioned above, this should be redundant.

File: docs/custom-flows/manage-sms-based-mfa.mdx (hunk 2 above)
Comment (alexisintech) [reply docs/custom-flows/manage-sms-based-mfa.mdx:62]: I liked having them grouped under their corresponding H2's. but I will move them 🤷‍♀️

**Comment by royanger** [comment]
Changes look good. I can't approve (my PR), but looks ready to merge.

**Review by alexisintech — APPROVED** [review]
YAY!!!

**Pull request merged**